### PR TITLE
Refactor private dns zone and zone group clients

### PR DESCRIPTION
**Pull request by lzhecheng**
* Use arm client
* Add UT

Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Refactor private dns zone and zone group clients
* Use arm client
* Add UT

related: #1493
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/azureclients/privatednsclient/azure_privatednsclient.go [hunk 1]
```diff
@@ -18,48 +18,216 @@ package privatednsclient
...
 	}
+
 	return client
 }
 
-// CreateOrUpdate creates or updates a private dns zone
-func (c *Client) CreateOrUpdate(ctx context.Context, resourceGroupName string, privateZoneName string, parameters privatedns.PrivateZone, waitForCompletion bool) error {
-	createOrUpdateFuture, err := c.privateDNSClient.CreateOrUpdate(ctx, resourceGroupName, privateZoneName, parameters, "", "*")
+// CreateOrUpdate creates or updates a private DNS zone.
+func (c *Client) CreateOrUpdate(ctx context.Context, resourceGroupName, privateDNSZoneName string, parameters privatedns.PrivateZone, etag string, wait bool) *retry.Error {
```
Comment (jwtty) [inline pkg/azureclients/privatednsclient/azure_privatednsclient.go:91]: Maybe keep the "waitForCompletion" name? Looks clearer to me.

File: pkg/azureclients/privatednszonegroupclient/interface.go [hunk 2]
```diff
@@ -20,15 +20,25 @@ import (
...
 	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-02-01/network"
+
+	"sigs.k8s.io/cloud-provider-azure/pkg/retry"
+)
+
+const (
+	// APIVersion is the API version.
+	APIVersion = "2021-02-01"
+	// AzureStackCloudAPIVersion is the API version for Azure Stack
+	AzureStackCloudAPIVersion = "2016-04-01"
```
Comment (jwtty) [inline pkg/azureclients/privatednszonegroupclient/interface.go:31]: which version should we use here?

File: pkg/azureclients/privatednsclient/azure_privatednsclient.go (hunk 1 above)
Comment (lzhecheng) [reply pkg/azureclients/privatednsclient/azure_privatednsclient.go]: Yes, we can do that.

File: pkg/azureclients/privatednszonegroupclient/interface.go (hunk 2 above)
Comment (lzhecheng) [reply pkg/azureclients/privatednszonegroupclient/interface.go]: Offline discussed. Will remove azure stack related code because it is not supported yet.

**Pull request merged**